### PR TITLE
[v2.14] Improve e2e tests stability

### PR DESCRIPTION
**Pull request by floreks**
Manual cherry-pick of #2501 changes.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by floreks** [comment]
It is more problematic than I expected. Tests against `v2.14` fail not because they are flaky, but because tests are using the latest API and due to changes in the API `v2.14` frontend will not work with the latest API.

It requires refactoring the way we run our tests to try and use proper API versions. This is a much bigger task, but I guess it is a high priority since we can't merge any fixes to `v2.14` now.

@kdomanski @kgroschoff

File: cypress/pages/members.po.ts [hunk 1]
```diff
@@ -1,6 +1,19 @@
-import {Group} from '../utils/member';
-import {wait} from '../utils/wait';
+// Copyright 2020 The Kubermatic Kubernetes Platform contributors.
```
Comment (floreks) [inline cypress/pages/members.po.ts:1]: Should license headers be there in v2.14?

Comment (kgroschoff) [reply cypress/pages/members.po.ts:1]: Probably yes, but seems like the [related PR](https://github.com/kubermatic/dashboard/pull/2454) was only merged into master 🤔

Comment (floreks) [reply cypress/pages/members.po.ts:1]: We can backport it then separately

**Comment by floreks** [comment]
Tests have passed 5 times in a row. I guess it is stable :slightly_smiling_face:

**Comment by kgroschoff** [comment]
🎉 
/lgtm

**Comment by floreks** [comment]
Added last small improvement to wait for project to be deleted before logging out.

**Pull request merged**